app/queues/dns/create-record-worker.server.ts

```
@@ -0,0 +1,142 @@
+import { prisma } from '~/db.server';
+import logger from '~/lib/logger.server';
+import { setMonthsFromNow } from '~/lib/domains.server';
+import { createRecord, getChangeStatus } from '~/lib/dns.server';
+import { RecordStatus } from '@prisma/client';
+
+import type { DomainRecord } from '~/lib/domains.server';
+
+export const addDbRecordQueueName = 'add-db-record';
+export const createDnsRecordQueueName = 'create-dns-record';
+export const checkDnsStatusQueueName = 'check-dns-status';
+export const syncDbStatusQueueName = 'sync-db-status';
+
+export const addDbRecordWorker = new Worker<
+  Pick<DomainRecord, 'username' | 'type' | 'name' | 'value'>
+>(
+  addDbRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in DB');
+    const { username, type, name, value } = job.data;
+
+    if (!username) {
+      throw new Error('Username is not provided');
+    }
+
+    const existingRecords = await prisma.record.count({
+      where: {
+        type,
+        name,
+        value,
+      },
+    });
+
+    if (existingRecords > 0) {
+      throw new Error('Record already exists');
+    }
+
+    try {
+      const result = await prisma.record.create({
+        data: {
+          username,
+          type,
+          name,
+          value,
+          expiresAt: setMonthsFromNow(6),
+          status: RecordStatus.pending,
+        },
+      });
+      return result.id;
+    } catch (error) {
+      logger.warn('Could not create a record in DB', { error });
+    }
+  },
+  { connection: redis }
+);
+
+export const createDnsRecordWorker = new Worker<
+  Pick<DomainRecord, 'username' | 'type' | 'name' | 'value'>
+>(
+  createDnsRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in Route 53');
+    const { username, type, name, value } = job.data;
+
+    try {
+      return await createRecord(username, type, name.toLowerCase(), value);
+    } catch (error) {
+      logger.warn('Could not create a record in Route53', { error });
```

please rethrow all errors, therwise this will only silently fail not stopping the process

---

Same comment on error interpolation with Winston.

---

Will be updated